exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/MurmurHash3.java

```diff
@@ -0,0 +1,264 @@
+    for (long i=start; i<roundedEnd; i+=16) {
+      long k1 = getLongLittleEndian(i);
+      long k2 = getLongLittleEndian(i+8);
+      k1 *= c1; k1  = Long.rotateLeft(k1,31); k1 *= c2; h1 ^= k1;
+      h1 = Long.rotateLeft(h1,27); h1 += h2; h1 = h1*5+0x52dce729;
+      k2 *= c2; k2  = Long.rotateLeft(k2,33); k2 *= c1; h2 ^= k2;
+      h2 = Long.rotateLeft(h2,31); h2 += h1; h2 = h2*5+0x38495ab5;
+    }
+
+    long k1 = 0;
+    long k2 = 0;
+
+    // tail
+    switch ((int)length & 15) {
+      case 15: k2  = (PlatformDependent.getByte(roundedEnd+14) & 0xffL) << 48;
+      case 14: k2 |= (PlatformDependent.getByte(roundedEnd+13) & 0xffL) << 40;
+      case 13: k2 |= (PlatformDependent.getByte(roundedEnd+12) & 0xffL) << 32;
+      case 12: k2 |= (PlatformDependent.getByte(roundedEnd+11) & 0xffL) << 24;
+      case 11: k2 |= (PlatformDependent.getByte(roundedEnd+10) & 0xffL) << 16;
+      case 10: k2 |= (PlatformDependent.getByte(roundedEnd+ 9) & 0xffL) << 8;
+      case  9: k2 |= (PlatformDependent.getByte(roundedEnd+ 8) & 0xffL);
+        k2 *= c2; k2  = Long.rotateLeft(k2, 33); k2 *= c1; h2 ^= k2;
+      case  8: k1  = (long)PlatformDependent.getByte(roundedEnd+7) << 56;
+      case  7: k1 |= (PlatformDependent.getByte(roundedEnd+6) & 0xffL) << 48;
+      case  6: k1 |= (PlatformDependent.getByte(roundedEnd+5) & 0xffL) << 40;
+      case  5: k1 |= (PlatformDependent.getByte(roundedEnd+4) & 0xffL) << 32;
+      case  4: k1 |= (PlatformDependent.getByte(roundedEnd+3) & 0xffL) << 24;
+      case  3: k1 |= (PlatformDependent.getByte(roundedEnd+2) & 0xffL) << 16;
+      case  2: k1 |= (PlatformDependent.getByte(roundedEnd+1) & 0xffL) << 8;
+      case  1: k1 |= (PlatformDependent.getByte(roundedEnd ) & 0xffL);
+        k1 *= c1; k1  = Long.rotateLeft(k1,31); k1 *= c2; h1 ^= k1;
+    }
+
+    h1 ^= length; h2 ^= length;
+
+    h1 += h2;
+    h2 += h1;
+
+    h1 = fmix64(h1);
+    h2 = fmix64(h2);
+
+    h1 += h2;
+    h2 += h1;
+    // murmur3_128 should return 128 bit (h1,h2), now we return only 64bits,
+    return h1;
+  }
+
+  public static long murmur3_64(long val, int seed) {
+
+    long h1 = seed & 0x00000000FFFFFFFFL;
+    long h2 = seed & 0x00000000FFFFFFFFL;
+
+    final long c1 = 0x87c37b91114253d5L;
+    final long c2 = 0x4cf5ad432745937fL;
+
+    int length = 8;
+    long k1 = 0;
+    long k2 = 0;
+    k1 = val;
+    k1 *= c1; k1  = Long.rotateLeft(k1,31); k1 *= c2; h1 ^= k1;
+
+    h1 ^= length; h2 ^= length;
+
+    h1 += h2;
+    h2 += h1;
+
+    h1 = fmix64(h1);
+    h2 = fmix64(h2);
+
+    h1 += h2;
+    h2 += h1;
+    // murmur3_128 should return 128 bit (h1,h2), now we return only 64bits,
+    return h1;
+
+  }
+
+  public static int murmur3_32(int bStart, int bEnd, DrillBuf buffer, int seed) {
+
+    final long c1 = 0xcc9e2d51L;
+    final long c2 = 0x1b873593L;
+    long start = buffer.memoryAddress() + bStart;
+    long end = buffer.memoryAddress() + bEnd;
+    long length = end - start;
```

Comment:
Why not use (bEnd - bStart) which would be subtracting smaller numbers..(instead of adding memoryAddress first and then subtracting)